app/models/reservation.rb

```diff
@@ -25,7 +25,7 @@ class Reservation < ActiveRecord::Base
   validate :not_in_past, :available, :check_banned, on: :create
 
   # correctly update the overdue flag if necessary
-  before_save :update_overdue, if: :checked_out?
+  before_save :update_overdue, if: :checked_out?, unless: self.changes.empty?
```

Comment:
If I had to guess I'd say it was an "and", but I'd have to look at the docs / source to be sure.